### ISSUE TITLE
Reduce functions don't work with Android Java Wrapper

### ISSUE THREAD

**Issue by vkkhare** [pytorch/pytorch#36484]
## 🐛 Bug

I seem to struggle with `reduce` functions in torchscript (ex: when I am using `mean()` or `sum()`). I am able to run these functions when the torchscript is run on python but when the same script is loaded into java wrapper I get **shape must not be empty** exactly at that location.

## To Reproduce

Steps to reproduce the behaviour in Kotlin:
1. [This](https://github.com/OpenMined/PySyft/blob/master/examples/experimental/FL%20Training%20Plan/Create%20Plan.ipynb) notebook can be used to generate the torchscript
2. Error in **mean()**
```
import torch as th
def cross_entropy_with_logits(output, target, batch_size):
    probs = th.nn.functional.softmax(output, dim=1)
    loss = -(target * th.log(probs)).mean()
    loss_grad = (probs - target) / (batch_size * target.shape[1])
    return probs, loss, loss_grad
```

3. [kotlin code gist](https://gist.github.com/vkkhare/9869be9ee9525fb9574f84d05356e723)

4. [Error Paste](https://pastebin.com/hfmDkceS)

## Expected behavior

Ideally, the code should have run without any errors and in fact if I use `th.cumsum(-(target * th.log(probs)),dim=1)` the torchscript runs perfectly fine.

## Environment
Collecting environment information...
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Ubuntu 19.10
GCC version: (Ubuntu 9.2.1-9ubuntu2) 9.2.1 20191008
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce 840M
Nvidia driver version: 430.50
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy==1.18.1
[pip3] torch==1.4.0
[pip3] torchvision==0.5.0

java pytorch version **1.4.0**

cc @suo

**Comment by suo** [comment]
@dreiss Do you mind taking a look? Related to the Java pytorch bindings.

**Comment by dreiss** [comment]
This was fixed in https://github.com/pytorch/pytorch/commit/6d1fa8296b847bb63185573d391bdb2cd3630b24 .  Please switch to the nightly build or the 1.5.0 release candidate.

As a workaround, the problem isn't related to reduction functions.  It's that the Java bindings in 1.4 cannot represent tensors with an empty shape.  So if you want to return such a tensor from your model, first call reshape.

```
cant_return = th.ones(1)[0]
print(cant_return)
print(cant_return.shape)
this_is_fine = cant_return.reshape(1)
print(this_is_fine)

tensor(1.)
torch.Size([])
tensor([1.])
```

**Comment by vkkhare** [comment]
Thanks I will try this out!

**Comment by vkkhare** [comment]
Reshape(1) works for me. I am using that for my purposes. Thanks
